core-plugins/docs/XMLReader-batchsource.md

```diff
@@ -0,0 +1,111 @@
+# XMLReader Batch Source
+
+
+Description
+-----------
+The XML Reader plugin is a source plugin that allows users to read XML files stored on HDFS.
+
+
+Use Case
+--------
+A user would like to read XML files that have been dropped into HDFS.
+These can be small to very large XML files. The XMLReader will read and parse the files,
+and when used in conjunction with the XMLParser plugin, fields can be extracted.
+This reader emits one XML event, specified by the node path property, for each file read.
+
+
+Properties
```

Comment:
It is not clear which of these properties are mandatory and which are optional. Incase they are optional would like to know what the default values are in description. Same change needs to be applied to the properties in Java code.